### PR TITLE
FEATURE: add a global setting to support custom docs url path

### PR DESCRIPTION
**Pull request by frank3manuel**
This PR adds a GlobalSetting `docs_path` to support custom docs URL path for sites that do not want docs page to live at `/docs` and have a customized path.

### CODE REVIEW

File: plugin.rb [hunk 1]
```diff
@@ -55,6 +55,15 @@
...
 
     any_user_agent[:disallow] ||= []
-    any_user_agent[:disallow] << "/docs/"
+    any_user_agent[:disallow] << "/#{SiteSetting.docs_url_path}/"
+  end
+
+  on(:site_setting_changed) do |name, old_value, new_value|
+    next if name != :docs_url_path
+
+    Discourse::Application.routes.append do
```
Comment (SamSaffron) [inline plugin.rb:64]: is this old code? feels like it is not safe

**Comment by SamSaffron** [comment]
OK this is getting super complicated and the constraint stuff is complicated too. I know I said it sounds like it can work but looking at the code I am scared.

I think we need to reboot. Lets just make this a GlobalSetting. That way most of the old code can remain and this becomes a feature on enterprise customers can consume (we change environment for them and redeploy) 

Having this per-site makes for complexity around the routing table if this is evaluated too early it slows down all routing.

File: plugin.rb (hunk 1 above)
Comment (frank3manuel) [reply plugin.rb:64]: Oh yes, that's old; will remove it.

**Comment by frank3manuel** [comment]
Looking back into this, using GlobalSetting.

**Comment by techAPJ** [comment]
I've updated the PR to clean up references to the SiteSetting, and remove obsolete route constraint.

File: assets/javascripts/discourse/templates/docs-topic-link.hbr [hunk 2]
```diff
@@ -1 +1 @@
-<a href="/docs?topic={{topic.id}}" class="docs-topic-link">{{{topic.fancyTitle}}}</a>
+<a href="/{{urlPath}}?topic={{topic.id}}" class="docs-topic-link">{{{topic.fancyTitle}}}</a>
```
Comment (pmusaraj) [inline assets/javascripts/discourse/templates/docs-topic-link.hbr:1]: Not a change in this PR, but does this URL starting with a `/` cause issues in a subfolder setup? Not necessary to fix in this PR, but would be good to know.

File: plugin.rb [hunk 3]
```diff
@@ -55,6 +57,10 @@
     end
 
     any_user_agent[:disallow] ||= []
-    any_user_agent[:disallow] << "/docs/"
+    any_user_agent[:disallow] << "/#{GlobalSetting.docs_path}/"
+  end
+
+  add_to_serializer(:site, :docs_path, "docs") do
```
Comment (pmusaraj) [inline plugin.rb:63]: The third parameter here is `define_include_method = true` by default and I'm not sure passing `"docs"` is helpful. 

Would be good to add a test for this serializer as well.

**Review by pmusaraj — APPROVED** [review]
Looks generally good, just a few minor comments. Also, would be good to add a PR description and a prefix (I'm guessing this is a `FEATURE: `?)

File: plugin.rb (hunk 3 above)
Comment (techAPJ) [reply plugin.rb]: Good catch! I've removed the incorrect `"docs"` and added a serializer test.

File: assets/javascripts/discourse/templates/docs-topic-link.hbr (hunk 2 above)
Comment (techAPJ) [reply assets/javascripts/discourse/templates/docs-topic-link.hbr:1]: I'll figure out a way to check this locally and will report back.

**Pull request merged**